modules/phase_field/src/kernels/ACInterface.C

```diff
@@ -41,7 +41,7 @@ ACInterface::ACInterface(const InputParameters & parameters)
   // Get mobility and kappa derivatives and coupled variable gradients
   for (unsigned int i = 0; i < _nvar; ++i)
   {
-    MooseVariable * ivar = _coupled_moose_vars[i];
+    MooseVariable * ivar = dynamic_cast<MooseVariable *>(_coupled_moose_vars[i]);
```

Comment:
in `ACSwitching` you used `MooseVariableFE` and did not have to use `dynamic_cast`. why could not you use it here?

Reply:
We call the `gradSln` method on `ivar` which isn't implemented for the abstract `MooseVariableFE`. We could do a dance where we get the `MooseVariable` using the `_subproblem` or `_sys` getters with `_coupled_moose_vars[i]->name()`. Or I could add another method(s) in `Coupleable` like `getCoupledStandardMooseVars` (and `getCoupledVectorMooseVars`) which would require adding a few more data members as well. I'm feeling lazy but will implement whatever design you think best